helm/chart/router/templates/deployment.yaml

```
@@ -26,6 +26,9 @@ spec:
       {{- end }}
       labels:
         {{- include "router.selectorLabels" . | nindent 8 }}
+        {{- if .Values.supergraphFile }}
+        checksum/supergraph: {{ .Values.supergraphFile | sha256sum | trunc 63 }}
```

```suggestion
        checksum/supergraph-config-map: {{ .Values.supergraphFile | sha256sum | trunc 63 }}
```
One other thing: I usually see this done as an annotation rather than as a label. Could you make that change?